components/skeleton/skeleton-mixin.js

```diff
@@ -187,42 +184,38 @@ export const SkeletonMixin = dedupeMixin(superclass => class extends RtlMixin(su
 
 	set skeleton(val) {
 		const oldVal = this._skeleton;
-		if (oldVal !== val) {
-			this._skeleton = val;
```

Comment:
I think we can do all the "work" in the setter and not rely on updated? This is now essentially saying: if we're not in a "wait to display" mode, just mirror _skeletonActive to _skeleton. Otherwise, tell the registry that something has changed and it'll call back to us if necessary.